### PR TITLE
Fix the D2K building selection decorations

### PR DESCRIPTION
**Pull request by pchote**
tags

This adds new "Primary Building" tag artwork for D2K (the original game repurposed the control group display, which doesn't work for us since we support grouped buildings), and implements the upgraded building decoration that the green star is supposed to be used for.

This is a simple pr that fixes a long standing polish annoyance, so i'd like to see this merged to prep.

The `WithDecoration:SelectionDecoration` option will be used in a future PR to generalize our horrible hardcoded tag and control group rendering.  This also provides our first shipped example of the global upgrades code.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Render/WithDecoration.cs [hunk 1]
```diff
@@ -87,6 +93,16 @@ public void PlaySingleFrame(int frame)
 
 		public IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
 		{
+			return !info.SelectionDecoration ? RenderInner(self, wr, self.Bounds) : Enumerable.Empty<IRenderable>();
+		}
+
+		public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr)
+		{
+			return info.SelectionDecoration ? RenderInner(self, wr, self.VisualBounds) : Enumerable.Empty<IRenderable>();
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Render/WithDecoration.cs]: This makes me thinkg `SelectionDecoration` is really a `should this be rendered at all`, not `only when selected`.

File: mods/d2k/rules/structures.yaml [hunk 2]
```diff
@@ -84,6 +84,17 @@ construction_yard:
...
+	GlobalUpgradable:
+		Prerequisites: upgrade.conyard
+		Upgrades: stardecoration
+	WithDecoration@upgraded:
+		SelectionDecoration: true
+		Image: pips
+		Sequence: tag-upgraded
+		ReferencePoint: Top, Right
+		ZOffset: 256
+		UpgradeTypes: stardecoration
```
Comment (Phrohdoh) [inline mods/d2k/rules/structures.yaml]: Why not `star-decoration`?

**Comment by Mailaender** [comment]
Works as promised. :white_check_mark: Indeed this was just a placeholder. Should be safe.

**Comment by penev92** [comment]
:+1:

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/f9719a50d60d2e6580f2dd64fed1823d060f9acf)

**Pull request merged**